dockstore-webservice/src/main/java/io/dockstore/webservice/jdbi/EntryDAO.java

```diff
@@ -398,13 +402,26 @@ private void processQuery(String filter, String sortCol, String sortOrder, Crite
                     query.orderBy(cb.asc(cb.size(entry.<Collection>get("starredUsers"))), cb.desc(entry.get("id")));
                 }
             } else {
-                Path<Object> sortPath = entry.get(sortCol);
-                if (!Strings.isNullOrEmpty(sortOrder) && "desc".equalsIgnoreCase(sortOrder)) {
-                    query.orderBy(cb.desc(sortPath), cb.desc(entry.get("id")));
-                    predicates.add(sortPath.isNotNull());
+                boolean hasSortCol = entry.getModel()
+                        .getAttributes()
+                        .stream()
+                        .map(Attribute::getName)
+                        .anyMatch(sortCol::equals);
+
+                if (!hasSortCol) {
+                    LOG.warn("Could not process query due to the invalid sortCol value.");
+                    throw new CustomWebApplicationException("Could not process query due to the invalid sortCol value.",
+                            HttpStatus.SC_BAD_REQUEST);
+
                 } else {
-                    query.orderBy(cb.asc(sortPath), cb.desc(entry.get("id")));
-                    predicates.add(sortPath.isNotNull());
+                    Path<Object> sortPath = entry.get(sortCol);
+                    if (!Strings.isNullOrEmpty(sortOrder) && "desc".equalsIgnoreCase(sortOrder)) {
+                        query.orderBy(cb.desc(sortPath), cb.desc(entry.get("id")));
+                        predicates.add(sortPath.isNotNull());
```

Comment:
Was like this before, but this line and line 423 look identical, so it could be done outside of the `if...else`